### DISCUSSION TITLE
Calling the create endpoint of a module (module1) inside the create endpoint of another module (module 2) given that they have different handlers

### DISCUSSION THREAD

**Discussion by ziaatrix** [serenity-is/Serenity#7142] [General]
Hi,

Any example how you are calling the create endpoint of a module (module1) inside the create endpoint of another module (module 2) given that they have different handlers.

![image](https://github.com/serenity-is/Serenity/assets/118987818/2b323a77-3691-4762-9a40-e41b7881a01d)

**Comment by ziaatrix** [comment]
I tried this way, but I could not initialize the handler of the module properly. Any idea?

![image](https://github.com/serenity-is/Serenity/assets/118987818/373cbbc8-44f4-41e2-ac3f-4c215d13cb6a)

**Comment by edwardch** [comment]
Try this

```cs
// add in the Create, just after the handler parameter 
[FromServices] IServiceResolver<IOtherSaveHandler> otherSaveHandlerResolver

// ..
var sr = new SaveRequest<OtherRow> { Entity = otherRecord };

otherSaveHandlerResolver.Resolve().Create(uow, sr);
```


hth

**Reply by ziaatrix** [reply]
Hi @edwardch ,

Thank you, this one works, however this otherSaveHandlerResolver is not returning the inserted id. I need as well to retrieve the inserted id after calling the Create endpoint.

**Reply by edwardch** [reply]
Not sure if this works, but something like 

```cs
var id = otherSaveHandlerResolver.Resolve().Create(uow, sr);
var x = id.EntityId;
```

**Reply by ziaatrix** [reply]
Hi @edwardch,

Actually this object does not have EntityId property.
![image](https://github.com/serenity-is/Serenity/assets/118987818/9eb57274-2a31-4c4d-a5db-3160d603289a)
